src/hardware/sblaster.cpp

```diff
@@ -1013,19 +1013,65 @@ static std::vector<AudioFrame>& maybe_silence(const T* samples,
 
 static uint32_t read_dma_8bit(const uint32_t bytes_to_read, const uint32_t i = 0)
 {
-	const auto bytes_read = sb.dma.chan->Read(bytes_to_read, sb.dma.buf.b8 + i);
-	assert(bytes_read <= DmaBufSize * sizeof(sb.dma.buf.b8[0]));
+	static_assert(sizeof(sb.dma.buf.b8[0]) == 1);
+
+	if (i >= DmaBufSize) {
+		// Should never happen as the code is current written.
+		// Calling code has i either 0 or 1 to handle a dangling sample from the last read.
```

Comment:
Could we rename the "i" arg to something more meaningful?

Reply:
fixed